tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -490,6 +490,28 @@ public function testIsUniqueMultipleFieldsOneIsNull()
     }
 
     /**
+     * Tests the IsNotReferencedBy rule before delete
+     *
+     * @group delete
+     * @return void
+     */
+    public function testIsNotReferencedByRule()
+    {
+        $articlesTable = TableRegistry::get('Articles');
+        $articlesTable->belongsTo('Authors');
+        $authorsTable = TableRegistry::get('Authors');
+
+        $author = $authorsTable->get(1);
+        $authorWithoutArticles = $authorsTable->get(2);
+
+        $rules = $authorsTable->rulesChecker();
+        $rules->addDelete($rules->IsNotReferencedBy('Articles'));
```

Comment:
You need to add a proxy method to `\Cake\ORM\RulesChecker` such as `public function isNotReferencedBy(array $tableOrAlias?, $message = null)`.

Also note that most (or all all) of the rules work on fields, so I would like to see that you can explicitly specify bindingKey and foreignKey (as array or array|string):

`public function isNotReferencedBy(array $tableOrAlias?, $message = null, $options)`